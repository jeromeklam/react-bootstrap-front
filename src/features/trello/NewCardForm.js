import React, { Component } from 'react';
import PropTypes from 'prop-types';

import EditableLabel from './EditableLabel';
import { Dropdown } from '../basic';

class NewCardForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      ref: React.createRef(),
      newLabel: '',
    };
    this.updateField = this.updateField.bind(this);
    this.projectSelect = this.projectSelect.bind(this);
    this.handleAdd = this.handleAdd.bind(this);
  }

  updateField = (field, value) => {
    this.setState({ [field]: value });
    if (field === 'label') {
      this.props.onPrjSearch(value);
    }
  };

  projectSelect = (prj) => {
    this.setState({ project: prj, newLabel: prj.label });
  };

  handleAdd = () => {
    this.props.onAdd(this.state);
  };

  render() {
    const { onCancel, t, projects } = this.props;
    return (
      <div className="trello-new-card-form">
        <article className="trello-new-card-form-wrapper">
          <header className="trello-new-card-form-header">
            <span className="trello-new-card-form-title">
              <EditableLabel
                placeholder={t({ id: 'rbf.trello.card.form.title.placeholder', defaultMessage: 'placeholder.title' })}
                onChange={val => this.updateField('title', val)}
                autoFocus
              />
            </span>
            <span className="trello-new-card-form-right" ref={this.state.ref}>
              <EditableLabel
                newLabel={this.state.newLabel}
                placeholder={t({ id: 'rbf.trello.card.form.label.placeholder', defaultMessage: 'placeholder.label' })}
                onKeyUp={e => this.updateField('label', e.target.value)}
              />
            </span>
            {projects && projects.length > 0 && (
              <Dropdown
                className="border rounded border-secondary bg-white text-secondary"
                myRef={this.state.ref}
                maxHeight="250px"
              >
                {projects.map(prj => (
                  <a
                    href="/"
                    key={`trello-card-${prj.id}`}
                    className="dropdown-item"
                    onClick={e => {
                      if (e) {
                        e.preventDefault();
                        e.stopPropagation();
                      }
                      this.projectSelect(prj);
                    }}
                  >
                    <span>{prj.label}</span>
                  </a>
                ))}
              </Dropdown>
            )}
          </header>
          <div className="rello-new-card-form-detail">
            <EditableLabel
              placeholder={t({
                id: 'rbf.trello.card.form.description.placeholder',
                defaultMessage: 'placeholder.description',
              })}
              onChange={val => this.updateField('description', val)}
            />
          </div>
        </article>
        <button
          className="btn btn-primary"
          title={t({ id: 'rbf.trello.card.form.add.help', defaultMessage: 'button.title.Add card' })}
          onClick={this.handleAdd}
        >
          {t({ id: 'rbf.trello.card.form.add', defaultMessage: 'button.Add card' })}
        </button>
        <button
          className="btn btn-secondary"
          title={t({ id: 'rbf.trello.card.form.cancel.help', defaultMessage: 'button.title.Cancel card' })}
          onClick={onCancel}
        >
          {t({ id: 'rbf.trello.card.form.cancel', defaultMessage: 'button.Cancel card' })}
        </button>
      </div>
    );
  }
}

NewCardForm.propTypes = {
  t: PropTypes.func.isRequired,
  onAdd: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  onPrjSearch: PropTypes.func.isRequired,
  projects: PropTypes.array,
};

NewCardForm.defaultProps = {
  projects: [],
};

export default NewCardForm;
