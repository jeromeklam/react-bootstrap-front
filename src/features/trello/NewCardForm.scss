.trello-new-card-form {
  background-color: #eee;
  margin-right: 15px;
}

.trello-new-card-form-wrapper {
  border-radius: 3px;
  background-color: #fff;
  position: relative;
  padding: 10px;
  cursor: pointer;
  max-width: 350px;
  margin-bottom: 7px;
  min-width: 330px;
  border: 1px solid;
  border-color: theme-color('secondary');
}

.trello-new-card-form-header {
  border-bottom: 1px solid #eee;
  padding-bottom: 6px;
  color: #000;
  margin-bottom: 10px;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
}

.trello-new-card-form-title {
  font-weight: normal;
  line-height: 18px;
  width: 70%;
  font-size: 14px;
}

.trello-new-card-form-right {
  width: 30%;
  text-align: right;
  padding-right: 10px;
  font-size: 10px;
}
.trello-new-card-form-detail {
  font-size: 12px;
  color: #4d4d4d;
  white-space: pre-wrap;
}