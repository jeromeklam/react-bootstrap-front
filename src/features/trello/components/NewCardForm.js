import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {
  CardForm,
  CardHeader,
  CardRightContent,
  CardTitle,
  CardWrapper,
  Detail
} from 'rt/styles/Base'
import {AddButton, CancelButton} from 'rt/styles/Elements'
import EditableLabel from 'rt/widgets/EditableLabel'

class NewCardForm extends Component {
  updateField = (field, value) => {
    this.setState({[field]: value})
  }

  handleAdd = () => {
    this.props.onAdd(this.state)
  }

  render() {
    const {onCancel, t} = this.props
    return (
      <CardForm>
        <CardWrapper>
          <CardHeader>
            <CardTitle>
              <EditableLabel placeholder={t('placeholder.title')} onChange={val => this.updateField('title', val)} autoFocus />
            </CardTitle>
            <CardRightContent>
              <EditableLabel placeholder={t('placeholder.label')} onChange={val => this.updateField('label', val)} />
            </CardRightContent>
          </CardHeader>
          <Detail>
            <EditableLabel placeholder={t('placeholder.description')} onChange={val => this.updateField('description', val)} />
          </Detail>
        </CardWrapper>
        <button className="btn btn-primary" onClick={this.handleAdd}>{t('button.Add card')}</button>
        <button className="btn btn-secondary" onClick={onCancel}>{t('button.Cancel')}</button>
      </CardForm>
    )
  }
}

NewCardForm.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onAdd: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
}

NewCardForm.defaultProps = {
}

export default NewCardForm
