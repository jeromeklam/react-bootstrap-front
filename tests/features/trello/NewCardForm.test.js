import React from 'react';
import renderer from 'react-test-renderer';
import NewCardForm from '../../../src/features/trello/NewCardForm';

const callback = jest.fn();
jest.useFakeTimers();

it('renders with className', () => {
  // First render
  const component = renderer.create(<NewCardForm t={callback} />);
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
