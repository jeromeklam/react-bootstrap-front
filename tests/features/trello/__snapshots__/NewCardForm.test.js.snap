// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with className 1`] = `
<div
  className="trello-new-card-form"
>
  <article
    className="trello-new-card-form-wrapper"
  >
    <header
      className="trello-new-card-form-header"
    >
      <span
        className="trello-new-card-form-title"
      >
        <div
          className="comPlainTextContentEditable comPlainTextContentEditable--has-placeholder"
          contentEditable="true"
          onBlur={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onPaste={[Function]}
          placeholder=""
        />
      </span>
      <span
        className="trello-new-card-form-right"
      >
        <div
          className="comPlainTextContentEditable comPlainTextContentEditable--has-placeholder"
          contentEditable="true"
          onBlur={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onPaste={[Function]}
          placeholder=""
        />
      </span>
    </header>
    <div
      className="rello-new-card-form-detail"
    >
      <div
        className="comPlainTextContentEditable comPlainTextContentEditable--has-placeholder"
        contentEditable="true"
        onBlur={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onPaste={[Function]}
        placeholder=""
      />
    </div>
  </article>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  />
  <button
    className="btn btn-secondary"
  />
</div>
`;
